implementing a limit_rows operation

tomreitz opened the pull request:
This PR implements a `limit_rows` operation. For example:
```yaml
      - operation: limit_rows
        count: 5 # required, no default
        offset: 10 # optional, default 0
```

jayckaiser reviewed (commented):
We should leave a comment that the `limit` operation will always return at _most_ `count` rows. `head()` and `tail()` can only retrieve rows from a single partition of a dataframe, which may be fewer rows than specified in `count`.

tomreitz commented:
> `head()` and `tail()` can only retrieve rows from a single partition of a dataframe

This is only true of `tail()`. `head()` [can](https://docs.dask.org/en/stable/generated/dask.dataframe.DataFrame.head.html) retrieve rows from any number of partitions via `npartitions=...`, up to all of them with `npartitions=-1`.

I've updated the PR to pass `head(..., npartitions=-1, ...)`, and I updated the README section to note that small dataframes may return fewer than `count` rows.
